website/server/libs/payments.js

```diff
@@ -321,4 +322,180 @@ api.buyGems = async function buyGems (data) {
   await data.user.save();
 };
 
+/**
+ * Allows for purchasing a user subscription, group subscription or gems with Stripe
+ *
+ * @param  options
+ * @param  options.token  The stripe token generated on the front end
+ * @param  options.user  The user object who is purchasing
+ * @param  options.gift  The gift details if any
+ * @param  options.sub  The subscription data to purchase
+ * @param  options.groupId  The id of the group purchasing a subscription
+ * @param  options.email  The email enter by the user on the Stripe form
+ * @param  options.headers  The request headers to store on analytics
+ * @return undefined
+ */
+api.payWithStripe = async function payWithStripe (options, stripeInc) {
+  let [
+    token,
+    user,
+    gift,
+    sub,
+    groupId,
+    email,
+    headers,
+  ] = options;
+  let coupon;
+  let response;
+  let subscriptionId;
+  // @TODO: We need to mock this, but curently we don't have correct Dependency Injection
+  let stripeApi = stripe;
```

Comment:
I'm pretty sure DI is preferred over `require-again`, but I'll take a look because what I have now is pretty bad.